lib/src/components/board/board.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:viam_sdk/src/resource/base.dart';
+
+import '../../gen/common/v1/common.pb.dart' as common;
+import '../../robot/client.dart';
+
+class BoardStatus {
+  final Map<String, int> analogs;
+  final Map<String, int> digitalInterrupts;
+
+  BoardStatus(this.analogs, this.digitalInterrupts);
+
+  factory BoardStatus.fromProto(common.BoardStatus pbBoardStatus) {
+    BoardStatus boardStatus = BoardStatus(Map<String, int>(), Map<String, int>());
+    pbBoardStatus.analogs.forEach((key, value) => boardStatus.analogs[key] = value.value);
+    pbBoardStatus.digitalInterrupts.forEach((key, value) => boardStatus.digitalInterrupts[key] = (value.value as int));
+    return boardStatus;
+  }
+}
+
+enum ViamPowerMode {
+  POWER_MODE_UNSPECIFIED,
+  POWER_MODE_NORMAL,
+  POWER_MODE_OFFLINE_DEEP,
+}
+
+/// Board represents a physical general purpose compute board that contains various
+/// components such as analog readers, and digital interrupts.
+///
+/// This acts as an abstract base class for any drivers representing specific
+/// board implementations. This cannot be used on its own.
+abstract class Board extends Resource {
+  static const Subtype subtype = Subtype(ResourceNamespaceRDK, ResourceTypeComponent, 'board');
+
+  @override
+  String name;
+
+  Board(this.name);
+
+  /// Get the status of the Board.
+  Future<BoardStatus> status({Map<String, dynamic>? extra});
+
+  /// Set the high/low state of the given pin of a board.
+  Future<void> setGPIO(String pin, bool high, {Map<String, dynamic>? extra});
```

Comment:
While I prefer having `GPIO` fully capitalized, I think it goes against the [flutter style guide](https://dart.dev/guides/language/effective-dart/style#do-capitalize-acronyms-and-abbreviations-longer-than-two-letters-like-words)

Reply:
Good callout, you're right. I think there might be a way to set up some static analysis for stuff like this too 👀